application/controllers/collection_instrument.py

```diff
@@ -87,42 +88,57 @@ def upload_to_bucket(self, exercise_id, file, ru_ref=None, classifiers=None, ses
         :param session: database session
         :return: a collection instrument instance
         """
-
-        log.info("Upload exercise", exercise_id=exercise_id)
+        log.info("Upload instrument", exercise_id=exercise_id)
 
         validate_uuid(exercise_id)
         self.validate_non_duplicate_instrument(file, exercise_id, session)
-        instrument = InstrumentModel(ci_type="SEFT")
-
-        seft_file = self._create_seft_file(instrument.instrument_id, file)
-        instrument.seft_file = seft_file
-
-        exercise = self._find_or_create_exercise(exercise_id, session)
-        instrument.exercises.append(exercise)
 
         survey = self._find_or_create_survey_from_exercise_id(exercise_id, session)
-        instrument.survey = survey
+        survey_id = survey.survey_id
+        survey_service_details = get_survey_service_details(survey_id)
+        ci_type = "SEFT"
+        instrument = InstrumentModel(ci_type=ci_type)
+        if classifiers:
+            classifiers = loads(classifiers)
+            if survey_service_details["surveyMode"] == "EQ_AND_SEFT":
+                self.validate_eq_and_seft_form_type(survey_id, classifiers.get("form_type"), ci_type, session)
+            instrument.classifiers = classifiers
 
+        exercise = self._find_or_create_exercise(exercise_id, session)
         if ru_ref:
             business = self._find_or_create_business(ru_ref, session)
             self.validate_one_instrument_for_ru_specific_upload(exercise, business, session)
             instrument.businesses.append(business)
 
-        if classifiers:
-            instrument.classifiers = loads(classifiers)
+        seft_file = self._create_seft_file(instrument.instrument_id, file)
+        instrument.seft_file = seft_file
+        instrument.exercises.append(exercise)
+        instrument.survey = survey
+        session.add(instrument)
 
         try:
-            survey_ref = get_survey_ref(instrument.survey.survey_id)
-            file.filename = survey_ref + "/" + exercise_id + "/" + file.filename
+            file.filename = survey_service_details["surveyRef"] + "/" + exercise_id + "/" + file.filename
             seft_ci_bucket = GoogleCloudSEFTCIBucket(current_app.config)
             seft_ci_bucket.upload_file_to_bucket(file=file)
         except Exception as e:
             log.exception("An error occurred when trying to put SEFT CI in bucket")
             raise e
 
-        session.add(instrument)
         return instrument
 
+    @staticmethod
+    def validate_eq_and_seft_form_type(survey_id: str, form_type: str, survey_mode: str, session: Session) -> None:
+        instruments = query_instruments_form_type_with_different_survey_mode(survey_id, form_type, survey_mode, session)
```

Comment:
A docstring on the function would help understand what this function is doing at a glance.  Something like 'This function ensures there are no existing instruments with a given formtype for a given survey'.

I know the error text is pretty clear why the failure is happening so it might be repeating yourself but I think it'd be useful.